template.yaml

```diff
@@ -1,68 +1,122 @@
-    SemanticVersion: 1.0.10
+    SemanticVersion: 1.1.0
     SourceCodeUrl: https://github.com/SumoLogic/sumologic-slack
     SpdxLicenseId: Apache-2.0
-Outputs:
-  SlackFunction:
-    Description: SlackScheduler Function ARN
-    Value:
-      Fn::GetAtt:
-      - SlackFunction
-      - Arn
+
+Conditions:
+  create_kms_key:
+    Fn::And:
+      - Fn::Equals:
+          - Ref: AwsKmsKeyARN
+          - ''
+      - Condition: create_secret
+  create_secret:
+    Fn::Equals:
+      - Ref: CreateSecret
+      - 'Yes'
+
 Parameters:
+  AwsKmsKeyARN:
+    Default: ''
+    Description: Please provide an existing KMS Key ARN to encrypt the Environment
+      variables HTTP_LOGS_ENDPOINT, TOKEN. If kept empty, a new KMS Key ID will be
+      created if CreateSecret is Yes.
+    Type: String
   BackfillDays:
     Type: String
-    Description: Enter the number of days for which logs should be collected. For eg, 2 means logs for last 2 days.
-    AllowedPattern: ".+"
+    Description: Enter the number of days for which logs should be collected. For
+      eg, 2 means logs for last 2 days.
+    AllowedPattern: .+
+  CreateSecret:
+    AllowedValues:
+      - 'Yes'
+      - 'No'
+    Default: 'No'
+    Description: Yes -> Encrypt the Provided Environment variables HTTP_LOGS_ENDPOINT,
+      TOKEN using KMS and Secret Manager. No -> No encryption.
+    Type: String
   HttpLogsEndpoint:
     Type: String
     Description: Paste the URL for the HTTP Logs source.
-    AllowedPattern: ".+"
+    AllowedPattern: .+
     NoEcho: true
   Token:
     Type: String
     Description: Paste the TOKEN generated from Slack App.
-    AllowedPattern: ".+"
+    AllowedPattern: .+
     NoEcho: true
   EnableInfrequentChannels:
     Type: String
-    Description: "True -> Enable dividing channels into frequent and infrequent based on the last message time.
-                  False -> Send all public channels messages."
+    Description: True -> Enable dividing channels into frequent and infrequent based
+      on the last message time. False -> Send all public channels messages.
     AllowedValues:
       - true
       - false
     Default: false
-  InfrequentChannelsThresholdInHours:
-    Type: String
-    Description: Threshold in hours to make channels as infrequent based on last message time. For eg, 12 hours means if the message is not recived for 12 hours, channel will be marked as infrequent.
-    Default: 72
-  InfrequentChannelsMessagesFetchTimeInHours:
-    Type: String
-    Description: Time in hours to fetch messages for InFrequent channels. For eg, 12 hours means send infrequent channels messages every 12 hours.
-    Default: 12
   DatabaseName:
     Type: String
-    Description: "DynamoDB DataBase Table Name."
+    Description: DynamoDB Table Name. Use the table name to identify the Slack Workspace for which you are collecting logs. Do not use the same DataBase Name across multiple installations.
     Default: sumoslack
+
 Resources:
+  LambdaRole:
```

Comment:
I am guessing since we are creating this role (instead of using default which has same assume role policy) we need it to reference it in the secret policy. Is Condition not supported in inline policy (under Policies parameter in SlackFunction)?

Reply:
Yes, the policy needs to be attached only when the secret is created